tools/testbench/common_test.c

```diff
@@ -281,3 +285,22 @@ void tb_enable_trace(bool enable)
 	else
 		debug_print("trace print disabled\n");
 }
+
+void tb_gettime(struct timespec *td)
+{
+#if !defined __XCC__
+	clock_gettime(CLOCK_MONOTONIC, td);
+#else
+	td->tv_nsec = 0;
+	td->tv_sec = 0;
+#endif
+}
+
+void tb_getcycles(uint64_t *cycles)
+{
+#if defined __XCC__
+	*cycles = XT_RSR_CCOUNT();
```

Comment:
it may be a int32 usage, no need int64.
https://elixir.bootlin.com/zephyr/v1.7.0-rc2/source/drivers/timer/xtensa_sys_timer.c#L41

Reply:
I'd keep this as 64 bits. With 400 MHz 32 bit counter overflows at 10.7s. Other Xtensas can be over 1 GHz clocked and we can support also other processor simulators too in the future.

Reply:
cycle should be a wrap data, once it go to max, then will go back to zero, so there is another issue with the code, 
please refer below:

	if (cycles1 > cycles0)
		diff = cycles1 - cycles0;
	else
		diff = UINT32_MAX - cycles0 + cycles1;

it is not a always increase data.

Reply:
> so there is another issue with the code,

What was the first issue?

> cycle should be a wrap data, once it go to max, then will go back to zero,

Yes and that's OK because unsigned integers (unlike signed integers) are guaranteed to wrap around by the standard

```python
import numpy as np
np.uint32(5) - np.uint32(0xffffffff)

6
```

https://www.acepace.net/integerQuiz/